rpc: Improve invalid vout value rpc error message

n1rna opened the pull request:
Since the `vout` value can start at `0`, the error message for *negative* values can be improved to something like: `vout cannot be negative`.

theStack commented:
Concept ACK -- "positive" means greater than zero by definition, so the error message was incorrect

I think you should squash those commits.

Welcome as new contributor by the way! :tada:

n1rna commented:
> Welcome as new contributor by the way!
THANKS @theStack

tryphe commented:
ACK 89860a8b52c7c84b987169ce238d8e8bc809ae7b. Agree with the concept, changes are no-risk.

theStack reviewed (approved):
ACK https://github.com/bitcoin/bitcoin/commit/89860a8b52c7c84b987169ce238d8e8bc809ae7b 🆗

adaminsky commented:
Concept ACK -- This error also appears some other places. Here are the first two I found from `git grep "vout must be positive"`, but there were some others.

https://github.com/bitcoin/bitcoin/blob/d052f5e6b79b9600f929457e9280b441b772785b/src/bitcoin-tx.cpp#L599-L600
https://github.com/bitcoin/bitcoin/blob/ffaac6e6142a1cf61e8e12d7fc406fcb02300144/src/rpc/rawtransaction_util.cpp#L177-L179

nima-fs commented:
> This error also appears some other places.
Will address those. @adaminsky

n1rna commented:
> This error also appears some other places.

Addressed. @adaminsky

adaminsky commented:
ACK 4be83808ba814dd5603dfcd72199ab723a4020f2.

MarcoFalke commented:
Could be a scripted diff to show the replacement is complete?

n1rna commented:
I was not familiar with scripted diff, I will replace my commit with with script diff enabled. @MarcoFalke

laanwj commented:
Please insert an empty line in your commit message between the title of the commit and the body, otherwise it appears like this in one-line summary format:
```
* f56bff455f6b965cd4f50ec8e6fe2bf5981ca30d scripted diff: Improve invalid vout value rpc error message -BEGIN VERIFY SCRIPT- r() { sed -i 's/vout must be positive/vout cannot be negative/g' $1 } r src/bitcoin-tx.cpp r src/rpc/rawtransaction_util.cpp r src/wallet.cpp r test/functional/rpc_rawtransaction.py -END VERIFY SCRIPT- (Nima Yazdanmehr) (pull/19956/head)
```

n1rna commented:
Addressed @laanwj

MarcoFalke commented:
Instead of hardcoding the file names, you can use `$(git grep -l 'vout must be positive')`

n1rna commented:
Yes, much better now. Thanks for the suggestoin @MarcoFalke

n1rna committed:
scripted diff: Improve invalid vout value rpc error message

-BEGIN VERIFY SCRIPT-
r() { sed -i 's/vout must be positive/vout cannot be negative/g' $1 }
r $(git grep -l 'vout must be positive')
-END VERIFY SCRIPT-

promag reviewed (commented):
Code review ACK f471a3be00c2b6433b8c258b716982c0539da13f.

fanquake reviewed (approved):
ACK f471a3be00c2b6433b8c258b716982c0539da13f
